src/flag_gems/ops/cummin.py

```diff
@@ -0,0 +1,341 @@
+import logging
+import math
+
+import torch
+import triton
+import triton.language as tl
+import triton.language.core as core
+
+from ..utils import libentry
+
+
+@triton.jit
+def _min_combine(a, idxa, b, idxb):
+    if a < b:
+        return a, idxa
+    else:
+        return b, idxb
```

Comment:
We can use tl.where here to avoid divergence.

```python
mask = a < b
out = tl.where(mask, a, b)
index = tl.where(mask, idxa, idxb)
```

Reply:
Or we can adopt from inductor here

```python
@triton.jit
def minimum_with_index(a_value, a_index, b_value, b_index):
    mask = a_value < b_value
    equal = a_value == b_value
    if is_floating(a_value):
        a_isnan = a_value != a_value
        b_isnan = b_value != b_value
        mask |= a_isnan and not b_isnan
        # Consider NaNs as equal
        equal |= a_isnan and b_isnan

    # Prefer lowest index if values are equal
    mask |= equal & (a_index < b_index)
    return tl.where(mask, a_value, b_value), tl.where(mask, a_index, b_index)
```

Reply:
We can add it to utis/triton_language_extension.py

Reply:
OK, I will change `minimum_with_index` a little, since the cummin need highest rather than lowest index if values are equal

Reply:
Yes, you are right